### PR TITLE
:book: plugins doc

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description**
Add plugins docs and how to use them. See the preview: https://deploy-preview-2115--kubebuilder.netlify.app/plugins/plugins.html

**Motivation**

Closes: https://github.com/kubernetes-sigs/kubebuilder/issues/2098

**Blocked by**
https://github.com/kubernetes-sigs/kubebuilder/pull/2106

### CODE REVIEW

File: docs/book/src/plugins/creating-plugins.md
Comment (lander2k2) [inline docs/book/src/plugins/creating-plugins.md:5]:
```suggestion
You can extend the Kubebuilder API to create your own plugins. If [extending the CLI][extending-cli], your plugin will be implemented in your project and registered to the CLI as has been done by the [SDK][sdk] project. See its [cli code][sdk-cli-pkg] as an example.
```

File: docs/book/src/plugins/creating-plugins.md
Comment (lander2k2) [inline docs/book/src/plugins/creating-plugins.md:9]:
```suggestion
Kubebuilder offers the Golang-based operator plugins, which will help its CLI tool users create projects following the [Operator Pattern][operator-pattern].
```

File: docs/book/src/plugins/creating-plugins.md
Comment (lander2k2) [inline docs/book/src/plugins/creating-plugins.md:11]:
```suggestion
The [SDK][sdk] project, for example, has language plugins for [Ansible][sdk-ansible] and [Helm][sdk-helm], which are similar options but for users who would like to work with these respective languages and stacks instead of Golang.
```

File: docs/book/src/plugins/creating-plugins.md
Comment (lander2k2) [inline docs/book/src/plugins/creating-plugins.md:13]:
```suggestion
Note that KubeBuilder provides the `config-base.kubebuilder.io` to help in these efforts. This plugin will scaffold the common base without any specific language scaffold file to allow you to extend the Kubebuilder style for your plugins.
```

File: docs/book/src/plugins/creating-plugins.md
Comment (lander2k2) [inline docs/book/src/plugins/creating-plugins.md:15]:
```suggestion
In this way, currently, you can [Extend the CLI][extending-cli] and use the `Bundle Plugin` to create your language plugins such as:
```

File: docs/book/src/plugins/creating-plugins.md
Comment (lander2k2) [inline docs/book/src/plugins/creating-plugins.md:24]:
```suggestion
If you do not want to develop your plugin using Golang, you can follow its standard by using the binary as follows:
```

File: docs/book/src/plugins/creating-plugins.md
Comment (lander2k2) [inline docs/book/src/plugins/creating-plugins.md:33]:
```suggestion
<h1>Why use the Kubebuilder style?</h1>
```

File: docs/book/src/plugins/creating-plugins.md
Comment (lander2k2) [inline docs/book/src/plugins/creating-plugins.md:35]:
```suggestion
Kubebuilder and SDK are both broadly adopted projects which leverage the [controller-runtime][controller-runtime] project.  They both allow users to build solutions using the [Operator Pattern][operator-pattern] and follow common standards.
```

File: docs/book/src/plugins/creating-plugins.md
Comment (lander2k2) [inline docs/book/src/plugins/creating-plugins.md:37]:
```suggestion
Adopting these standards can bring significant benefits, such as joining forces on maintaining Kubebuilder and the common standards maintained there.  This allows you to focus on the specific needs and requirements for your plugin and use-case.
```

File: docs/book/src/plugins/creating-plugins.md
Comment (lander2k2) [inline docs/book/src/plugins/creating-plugins.md:39]: Unsure what we're saying here.  Have we already covered this idea in the previous two sentences?  If so, perhaps we should remove this?

File: docs/book/src/plugins/creating-plugins.md
Comment (lander2k2) [inline docs/book/src/plugins/creating-plugins.md:45]:
```suggestion
Note that users are also able to use plugins to customize their scaffold and address specific needs. See that Kubebuilder provides the [declarative][declarative-code] plugin which can be used when for example an API is scaffold:
```
Is this the intended link?

**Review by lander2k2 — COMMENTED** [review]
Added some suggestions - mostly word-smithing and link-fixing.  Two general points:
1. Would the "Project config" fit better in the "Reference" section since it applies to more than just plugins?
2. There is a "CLI Plugins" section (5.12)... should that be moved to the new "Plugins" section?

**Comment by lander2k2** [comment]
@camilamacedo86 These plugin docs are really helpful - thanks for adding them.  Something that would also be super-helpful is a more complete example of a plugin that does something trivial like modify the base image in the Dockerfile.  It wasn't immediately apparent to me how to go about this from looking through the SDK source.  If I can get something working, I'll be happy to help by adding something along these lines later on.

File: docs/book/src/plugins/creating-plugins.md
Comment (camilamacedo86) [reply docs/book/src/plugins/creating-plugins.md:39]: I want to make clear that those who decided to adopt this standard will able not to;  take advantage of what is provided by KB/SDK now and in the future. Also, this person can use any other solution that might came to exist as well which is respecting that.

File: docs/book/src/plugins/project-config.md
Comment (camilamacedo86) [reply docs/book/src/plugins/project-config.md:40]: It is the correct value. See: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v3/PROJECT#L25

Comment (camilamacedo86) [reply docs/book/src/plugins/project-config.md:40]: I fixed it on the table. Tks.

File: docs/book/src/plugins/creating-plugins.md
Comment (camilamacedo86) [reply docs/book/src/plugins/creating-plugins.md:39]: I will close that as addressed however, please feel free to re-open if you see that the text is not clear. Also, please feel free to raise suggestions for a better re-phase too.

**Comment by camilamacedo86** [comment]
Hi @lander2k2, 

I really like the idea of the small example. We can add a mini tutorial in this section as well. WDYT? Would you like to work on that as follow up of this one?

**Review by djzager — COMMENTED** [review]
A couple comments, other than that it looks good.

File: docs/book/src/plugins/creating-plugins.md
Comment (camilamacedo86) [reply docs/book/src/plugins/creating-plugins.md:13]: solved

**Comment by camilamacedo86** [comment]
/hold 

Only too give the chance of we get more reviews before the merge.

**Review by djzager — APPROVED** [review]

File: docs/book/src/plugins/extending-cli.md
Comment (lander2k2) [inline docs/book/src/plugins/extending-cli.md:23]: @camilamacedo86 Sorry I didn't bring this up earlier... but I just discovered this while building a test plugin.  This pkg doesn't seem to exist.  Is this import path right?  I'm looking in your branch for PR #2106 : https://github.com/camilamacedo86/kubebuilder/tree/config-base/pkg/plugins

Comment (camilamacedo86) [reply docs/book/src/plugins/extending-cli.md:23]: We changed the path and I did not updated that here. It is : `kustomizecommonv1 "sigs.k8s.io/kubebuilder/v3/pkg/plugins/common/kustomize/v1"`

Comment (camilamacedo86) [reply docs/book/src/plugins/extending-cli.md]: Done now 👍

File: docs/book/src/plugins/extending-cli.md
Comment (camilamacedo86) [reply docs/book/src/plugins/extending-cli.md]: done

File: docs/book/src/plugins/extending-cli.md
Comment (camilamacedo86) [reply docs/book/src/plugins/extending-cli.md:183]: done

**Review by djzager — APPROVED** [review]
/lgtm

**Comment by camilamacedo86** [comment]
/hold cancel

As an initial doc for the plugins that seem great enough.
I am looking for a great idea of the tutorial as follow up @lander2k2 please feel free to ping me. :-)

**Pull request merged**
